resource-manager/src/main/java/org/snomed/module/storage/ModuleStorageCoordinator.java

```diff
@@ -0,0 +1,602 @@
+     * @return ModuleMetadata stored for given arguments.
+     */
+    public Optional<ModuleMetadata> download(String codeSystem, String moduleId, String effectiveTime) {
+        return download(codeSystem, moduleId, effectiveTime, true);
+    }
+
+    /**
+     * Download ModuleMetadata stored for given arguments.
+     *
+     * @param codeSystem    CodeSystem of RF2 package.
+     * @param moduleId      Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime Effective time of RF2 package.
+     * @param includeFile   Whether to download RF2 package.
+     * @return ModuleMetadata stored for given arguments.
+     */
+    public Optional<ModuleMetadata> download(String codeSystem, String moduleId, String effectiveTime, boolean includeFile) {
+        if (invalidArguments(codeSystem, moduleId, effectiveTime)) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.ofNullable(doDownloadRelease(codeSystem, moduleId, effectiveTime, includeFile));
+        } catch (Exception e) {
+            LOGGER.error("Cannot download ModuleMetadata; exception encountered", e);
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Archive ModuleMetadata stored for given arguments by moving appropriate files to an "archive" subdirectory.
+     *
+     * @param codeSystem    CodeSystem of RF2 package.
+     * @param moduleId      Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime Effective time of RF2 package.
+     * @return Boolean indicating success of operation.
+     */
+    public boolean archive(String codeSystem, String moduleId, String effectiveTime) {
+        if (!allowArchive || invalidArguments(codeSystem, moduleId, effectiveTime)) {
+            return false;
+        }
+
+        try {
+            return doArchiveRelease(codeSystem, moduleId, effectiveTime);
+        } catch (Exception e) {
+            LOGGER.error("Cannot archive; exception encountered", e);
+            return false;
+        }
+    }
+
+    /**
+     * Download ModuleMetadata stored for given arguments, as well as optionally downloading dependent ModuleMetadata.
+     *
+     * @param codeSystem          CodeSystem of RF2 package.
+     * @param moduleId            Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime       Effective time of RF2 package.
+     * @param includeFile         Whether to download RF2 package.
+     * @param includeDependencies Whether to download dependent ModuleMetadata.
+     * @return Collection of dependent ModuleMetadata, sorted by effective time.
+     */
+    public List<ModuleMetadata> downloadRelease(String codeSystem, String moduleId, String effectiveTime, boolean includeFile, boolean includeDependencies) {
+        if (invalidArguments(codeSystem, moduleId, effectiveTime)) {
+            return Collections.emptyList();
+        }
```

Comment:
@pgwilliams `download` downloads an individual ModuleMetadata item, whereas `downloadRelease` downloads the packages entire dependency tree.